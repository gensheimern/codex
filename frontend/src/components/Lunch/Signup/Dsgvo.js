/*
*@author Stella Neser & Diana Beldiman 
* Komponente erzeugt ein Dialogfeld für die Datenschutzgrundverordnung(Dsgvo)
* Wird bei der Registrierung nach aufrufen angezeigt
* Komponente lädt den Text der Dsgvo aus DsgvoText.js 
*/
import React from 'react';
import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';
import RaisedButton from 'material-ui/RaisedButton';
import {RadioButton, RadioButtonGroup} from 'material-ui/RadioButton';
import CreateDsgvoText from '../signup/DsgvoText';

const styles = {
    radioButton: {
      marginTop: 16,
    },
  };

/**
 * Dialog content can be scrollable.
 */
export default class Dsgvo extends React.Component{
    constructor(props) {
        super(props);

        this.state = {
            open: false,
        };
    }
      handleOpen = () => {
        this.setState({open: true});
      };

      handleClose = () => {
        this.setState({open: false});
      };

      handleClick = () => {
        this.props.closeDrawer();
        this.handleOpen();
      };

      
    render(){
        const actions = [
            <FlatButton
              label="Back"
              primary={true}
              onClick={this.handleClose}
            />,
        ];     
        
        return(
        <div>
            <FlatButton className="flatButton1" label="read here" onClick={this.handleOpen}/>
            <Dialog
            title="Datenschutzerklärung"
            actions={actions}
            modal={false}
            open={this.state.open}
            onRequestClose={this.handleClose}
            autoScrollBodyContent={true}
            >
           <p>{CreateDsgvoText.dsgvo}</p>
            </Dialog>
        </div>        
        );
    }
}
