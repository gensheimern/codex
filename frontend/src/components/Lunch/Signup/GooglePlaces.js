import React from 'react';
import PlacesAutocomplete, { geocodeByAddress } from 'react-places-autocomplete';
import TextField from 'material-ui/TextField';

export default class LocationSearchInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      address: '',
    }
    this.parseAddress = this.parseAddress.bind(this);
  }

  handleChange = (address) => {
    this.setState({ address });
  }

  parseAddress(results){
    let parsedAddress = results.map(e =>{
      return(
      e.address_components.map(f => {
           if( f.types[0] === 'street_number' ||  f.types[0] ==='route' || f.types[0] === 'locality'){
              return(this.props.myAddress(f))
          }
          return true;
      }))

    })
    console.log(parsedAddress);
  }

  handleSelect = (address) => {
    this.props.callbackAdress(address);
    geocodeByAddress(address)
      .then(results => this.props.myAddress(results[0]))
      .then(latLng => console.log('Success', latLng))
      .catch(error => console.error('Error', error))
  }

  render() {
    return (
      <PlacesAutocomplete
        value={this.state.address}
        onChange={this.handleChange}
        onSelect={this.handleSelect}
        maxSearchResults={20}
      >
        {({ getInputProps, suggestions, getSuggestionItemProps }) => (
          <div>
          <TextField
                style={this.props.styles}
                floatingLabelFocusStyle={{color:"rgb(30 161 133)"}}
                underlineFocusStyle={{borderColor:"rgb(30 161 133)"}}
                floatingLabelFixed={true}
                floatingLabelText="Address"
                hintText="Search your location"
                {...getInputProps({
                  className: 'location-search-input'
                })}
          /><br />

            <div className="autocomplete-dropdown-container">
              {suggestions.map(suggestion => {
                const className = suggestion.active ? 'suggestion-item--active' : 'suggestion-item';
                // inline style for demonstration purpose
                const style = suggestion.active
                            ? { backgroundColor: '#fafafa', cursor: 'pointer', marginTop:"10px" }
                            : { backgroundColor: '#ffffff', cursor: 'pointer', marginTop:"10px" };
                return (
                  <div {...getSuggestionItemProps(suggestion, { className, style })}>
                    <span>{suggestion.description}</span>
                  </div>
                )
              })}
            </div>
          </div>
        )}
      </PlacesAutocomplete>
    );
  }
}
