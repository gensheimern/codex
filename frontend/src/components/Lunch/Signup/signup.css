.signupR {
	padding: 2%;
	/*height: 1065px;*/
	height: 100vh;
	width: 100vw;
	/*opacity: 0.87;*/
	/*z-index: -1;*/
	overflow-y: scroll;
}

.signupBgR {
	background-image: url("Hintergrundbild.png");
	background-size: cover;
	background-repeat: no-repeat;
	background-color: #1EA185;
	background-position: center top;
}

.h3headerR {
	font-family: 'Trebuchet MS';
	font-size: 30px;
	font-weight: 'bold';
	line-height: 35px;
	color: #FFFFFF;
	padding-bottom: 30px;
}

.h2headerR {
	margin: 10% 10% 0 10%;
	color: #1EA185;
	font-family: 'Trebuchet MS';
	font-size: 20px;
	font-weight: bold;
	line-height: 23px;
	text-align: left;
}

.signupCardR {
	width: 440px;
	border-radius: 5px;
	background-color: #FFFFFF;
	display: block;
	margin-left: auto;
	margin-right: auto;
	text-align: center;
}

.toggleWrapperR {
	float:left;
	width:80%;
	padding-top:20px;
	margin-left:10%;
	margin-right:10%;
}


@media only screen and (max-width: 500px) {
    .signupCardR {
		width: 100%;
    }
}
