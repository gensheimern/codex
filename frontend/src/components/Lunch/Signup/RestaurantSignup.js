import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';
import config from '../../../config';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import {Card, /*CardMedia, CardTitle,*/ CardText} from 'material-ui/Card';
import './signup.css';
import Paper from 'material-ui/Paper';
import Checkbox from 'material-ui/Checkbox';
import Maps from './GooglePlaces';
import {Link} from 'react-router-dom';
import Toggle from './SignupToggle';

export default class Signup extends React.Component {
	emailRegExp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

	//Recaptcha = require('react-recaptcha');

	constructor(props) {
		super(props);

		this.handleChange.bind(this);
		this.signupUser.bind(this);
		this.callbackAddress = this.callbackAddress.bind(this);
		this.callbackToggle = this.callbackToggle.bind(this);

		this.state = {
			email: "",
			restaurantName: "",
			googlePlacesId: '',
			place: '',
			zipcode: '',
			street: '',
			streetNumber: '',
			password: "",
			retypePassword: "",
			checked: false,
			captcha: '',
			addressValue: '',
			toggle: true,
			signupGoogleSearch: true,
		}
	}

	handleChange = name => e => {
		this.setState({
			[name]: e.target.value
		});
	}

	callbackToggle(event, isInputChecked){
		this.setState({ signupGoogleSearch: isInputChecked });
	}

	validateForm() {
		return this.emailRegExp.test(this.state.email)
			&& this.state.restaurantName !== ""
			&& this.state.googlePlacesId !== ""
			&& this.state.password !== ""
			&& this.state.retypePassword !== ""
			&& this.state.password === this.state.retypePassword
			&& this.state.checked
	}

	updateCheck() {
		this.setState((oldState) => {
		  return {
			checked: !oldState.checked,
		  };
		});
	}

	callbackAddress(myAddress){
		this.setState({ googlePlacesId: myAddress.split(',') });
	}

	handleChangeAddressValue(e){
		this.setState({ addressValue: e.target.value });
	}

	toggle(){

	}

	signupUser = (e) => {
		e.preventDefault();

		fetch(config.apiPath + "/restaurant", {
			method: 'POST',
			body: JSON.stringify({
					googlePlacesId: this.state.googlePlacesId,
	        name: this.state.restaurantName,
	        email: this.state.email,
	        password: this.state.password,
	        place: this.state.place,
	        zipcode: this.state.zipcode,
	        street: this.state.street,
	        streetNumber: this.state.streetNumber
			}),
			headers: {
				'Content-Type': 'application/json',
			}
		})
		.then((res) => {
			if (!res.ok || res.status !== 201) {
				// handle error
			} else {

				this.props.history.push('/feed');
			}
		});
	}

	render() {
		const styles = {
			block: {
			  maxWidth: 250,
			},
			checkbox: {
				paddingLeft: 100,
				paddingRight: 197,
				marginBottom: 16,
				fontWeight: 0,
			},

			cardStyle:{
				width: 440,
				borderRadius: 5,
				backgroundColor: "#FFFFFF",
				boxShadow: "0 2 20 0 rgba(57,56,56,0.5)",
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',
			},

			paperStyle:{
				height: 62,
				width: 62,
				borderRadius: 7,
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',
			},
			textField:{
				width: '80%',
				marginLeft: '10%',
				marginRight: '10%',
			},
			reCAPTCHA:{
				heigth: 76,
				width: 305,
				boarder:"1 solid #D6D6D6",
				borderRadius: 2.43,
				backgroundColor: "#FAFAFA",
				boxShadow: "0 1 2 0 rgba(0,0,0,0.1)",
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',

			},
			signupButton:{
				heigt: 40.57,
				width: '30%',
				borderRadius: 3,
				boxShadow: "inset 0 1 3 0 rgba(0,0,0,0.5),0 1 2 0 rgba(0,0,0,0.5)",
				backgroundColor: "#1EA185",
				display: 'block',
				marginLeft: 'auto',
				marginRight: 'auto',
			},


		};

		let retypeError = {};
		if (this.state.password !== this.state.retypePassword) {
			retypeError = {
				errorText: 'Passwords do not match.',
			};
		}


		if(this.state.signupGoogleSearch){
			var content =
			<Maps
				styles={styles.textField}
				callbackAdress={this.handleChangeAddressValue}
				myAddress={this.callbackAddress}
				>
				{renderFunc}
			</Maps>

		}
		else {
			var content =
			<div>
			<TextField
				id="restaurantName"
				label="restaurantName"
				value={this.state.restaurantName}
				onChange={this.handleChange('restaurantName')}
				floatingLabelText="Restaurant Name"
				style= {styles.textField}
			/>



			<TextField
				id="googlePlacesId"
				label="googlePlacesId"
				value={this.state.googlePlacesId}
				onChange={this.handleChange('googlePlacesId')}
				floatingLabelText="Find your Restaurant"
				style= {styles.textField}
			/>

			<TextField
				id="place"
				label="Place"
				value={this.state.place}
				onChange={this.handleChange('place')}
				floatingLabelText="Place"
				style= {styles.textField}
			/>
			<TextField
				id="zipcode"
				label="Zipcode"
				value={this.state.zipcode}
				onChange={this.handleChange('zipcode')}
				floatingLabelText="Zipcode"
				style= {styles.textField}
			/>
			<TextField
				id="street"
				label="Street"
				value={this.state.street}
				onChange={this.handleChange('street')}
				floatingLabelText="Street"
				style= {styles.textField}
			/>
			<TextField
				id="streetNumber"
				label="Street number"
				value={this.state.streetNumber}
				onChange={this.handleChange('streetNumber')}
				floatingLabelText="Street number"
				style= {styles.textField}
			/>
			<TextField
				id="email"
				label="E-Mail"
				value={this.state.email}
				onChange={this.handleChange('email')}
				floatingLabelText="E-Mail"
				style= {styles.textField}
			/>

			<TextField
				id="password"
				label="Password"
				type="password"
				value={this.state.password}
				onChange={this.handleChange('password')}
				floatingLabelText="Password"
				style= {styles.textField}
			/>

			<TextField
				id="retypePassword"
				label="Retype password"
				type="password"
				value={this.state.retypePassword}
				onChange={this.handleChange('retypePassword')}
				floatingLabelText="Retype Password"
				{ ...retypeError }
				style= {styles.textField}
			/>
		</div>
		}



		return(
			<div className="signupBg">
			<form className="signup" onSubmit={this.signupUser}>
				<div>
					<Paper style={styles.paperStyle} zDepth={1} />
 				</div>

				<br/>
				<center>
					<h3 className="h3header">Lunchplanner</h3>
				</center>

				<Card className="signupCard">
				<CardText>
				<h2 className="h2header">SIGN UP FOR YOUR RESTAURANT</h2>

				<div className="toggleWrapper">
						<label style={{float:"left", width:"50%",paddingLeft:"10%"}}> Google Search </label>
						<Toggle toggle={this.callbackToggle}  />
				</div>
				{content}
				<br/>
				<br/>

				<Checkbox
          			label="DatenschutzerklÃ¤rung "
          			checked={this.state.checked}
          			onCheck={this.updateCheck.bind(this)}
          			style={styles.checkbox}
        		/>

				<RaisedButton
					backgroundColor="#b9b9b9"
					type="submit"
					label="Sign Up"
					primary={true}
					disabled={!this.validateForm()}
					style = {styles.signupButton}
				/>
				<br/>
				<center>
					<p>Have an account already?&nbsp;
						<Link to = "/login">Log in here </Link>
					</p>
				</center>

				</CardText>
				</Card>

			</form>
			</div>

		);
	}
}
const renderFunc = ({ getInputProps, getSuggestionItemProps, suggestions }) => (
	<div className="autocomplete-root">
		<input {...getInputProps()} />
		<div className="autocomplete-dropdown-container">
			{suggestions.map(suggestion => (
				<div {...getSuggestionItemProps(suggestion)}>
					<span>{suggestion.description}</span>
				</div>
			))}
		</div>
	</div>
);
