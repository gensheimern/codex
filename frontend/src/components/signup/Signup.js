import React from 'react';
import config from '../../config';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import {Card, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import ImageBackground from 'react';
import './signup.css';
import Avatar from 'material-ui/Avatar';

export default class Signup extends React.Component {
	emailRegExp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

	constructor(props) {
		super(props);

		this.handleChange.bind(this);
		this.signupUser.bind(this);

		this.state = {
			email: "",
			name: "",
			firstName: "",
			password: "",
			retypePassword: "",
			img: "",
		}
	}

	handleChange = name => e => {
		this.setState({
			[name]: e.target.value
		});
	}

	validateForm() {
		return this.emailRegExp.test(this.state.email)
			&& this.state.firstName !== ""
			&& this.state.name !== ""
			&& this.state.password !== ""
			&& this.state.retypePassword !== ""
			&& this.state.password === this.state.retypePassword;
	}

	signupUser = (e) => {
		e.preventDefault();

		fetch(config.apiPath + "/user", {
			method: 'POST',
			body: JSON.stringify({
				firstName: this.state.firstName,
				name: this.state.name,
				email: this.state.email,
				password: this.state.password,
				image: this.state.img,
			}),
			headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken')
			}
		});
	}

	render() {
		let retypeError = {};
		if (this.state.password !== this.state.retypePassword) {
			retypeError = {
				errorText: 'Passwords do not match.',
			};
		}
		return(

			<form className="signup" onSubmit={this.signupUser}>
			<div>
 <Avatar alt="" src="#"  />
 </div>
<Card>
<CardText>
				<TextField
					id="firstName"
					label="first name"
					value={this.state.firstName}
					onChange={this.handleChange('firstName')}
					floatingLabelText="First name"
				/>
				<br/>

				<TextField
					id="name"
					label="Last name"
					value={this.state.name}
					onChange={this.handleChange('name')}
					floatingLabelText="Last name"
				/>
				<br/>

				<TextField
					id="email"
					label="E-Mail"
					value={this.state.email}
					onChange={this.handleChange('email')}
					floatingLabelText="E-Mail"
				/>
				<br/>

				<TextField
					id="password"
					label="Password"
					type="password"
					value={this.state.password}
					onChange={this.handleChange('password')}
					floatingLabelText="Password"
				/>
				<br/>

				<TextField
					id="retypePassword"
					label="Retype password"
					type="password"
					value={this.state.retypePassword}
					onChange={this.handleChange('retypePassword')}
					floatingLabelText="Retype Password"
					{ ...retypeError }
				/>
				<br/>
				<br/>

				<RaisedButton
					type="submit"
					label="Sign Up"
					primary={true}
					disabled={!this.validateForm()}
				/>
				</CardText>
				</Card>
            </form>

		);
	}
}
