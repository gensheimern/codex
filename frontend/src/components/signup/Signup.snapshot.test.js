import React from 'react';
import renderer from'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';

import Signup from './Signup';

describe('Signup Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <Signup />
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});