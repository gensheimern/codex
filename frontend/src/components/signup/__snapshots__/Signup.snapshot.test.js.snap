// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Snapshot renders 1`] = `
<form
  className="signup"
  onSubmit={[Function]}
>
  <h1>
    Sign Up
  </h1>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 72,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <label
      className={undefined}
      htmlFor="firstName"
      onClick={undefined}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "lineHeight": "22px",
          "muiPrepared": true,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 38,
          "transform": "scale(1) translate(0, 0)",
          "transformOrigin": "left top",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "zIndex": 1,
        }
      }
    >
      First name
    </label>
    <input
      disabled={false}
      id="firstName"
      label="first name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "marginTop": 14,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <br />
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 72,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <label
      className={undefined}
      htmlFor="name"
      onClick={undefined}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "lineHeight": "22px",
          "muiPrepared": true,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 38,
          "transform": "scale(1) translate(0, 0)",
          "transformOrigin": "left top",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "zIndex": 1,
        }
      }
    >
      Last name
    </label>
    <input
      disabled={false}
      id="name"
      label="Last name"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "marginTop": 14,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <br />
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 72,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <label
      className={undefined}
      htmlFor="email"
      onClick={undefined}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "lineHeight": "22px",
          "muiPrepared": true,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 38,
          "transform": "scale(1) translate(0, 0)",
          "transformOrigin": "left top",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "zIndex": 1,
        }
      }
    >
      E-Mail
    </label>
    <input
      disabled={false}
      id="email"
      label="E-Mail"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "marginTop": 14,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <br />
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 72,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <label
      className={undefined}
      htmlFor="password"
      onClick={undefined}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "lineHeight": "22px",
          "muiPrepared": true,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 38,
          "transform": "scale(1) translate(0, 0)",
          "transformOrigin": "left top",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "zIndex": 1,
        }
      }
    >
      Password
    </label>
    <input
      disabled={false}
      id="password"
      label="Password"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "marginTop": 14,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="password"
      value=""
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <br />
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "transparent",
        "cursor": "auto",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "height": 72,
        "lineHeight": "24px",
        "muiPrepared": true,
        "position": "relative",
        "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": 256,
      }
    }
  >
    <label
      className={undefined}
      htmlFor="retypePassword"
      onClick={undefined}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.3)",
          "lineHeight": "22px",
          "muiPrepared": true,
          "pointerEvents": "none",
          "position": "absolute",
          "top": 38,
          "transform": "scale(1) translate(0, 0)",
          "transformOrigin": "left top",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "zIndex": 1,
        }
      }
    >
      Retype Password
    </label>
    <input
      disabled={false}
      id="retypePassword"
      label="Retype password"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      style={
        Object {
          "WebkitOpacity": 1,
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "appearance": "textfield",
          "backgroundColor": "rgba(0,0,0,0)",
          "border": "none",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "inherit",
          "font": "inherit",
          "height": "100%",
          "marginTop": 14,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
      type="password"
      value=""
    />
    <div>
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 1,
            "borderColor": "#e0e0e0",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "width": "100%",
          }
        }
      />
      <hr
        aria-hidden="true"
        style={
          Object {
            "borderBottomStyle": "solid",
            "borderBottomWidth": 2,
            "borderColor": "#00bcd4",
            "borderLeft": "none",
            "borderRight": "none",
            "borderTop": "none",
            "bottom": 8,
            "boxSizing": "content-box",
            "margin": 0,
            "muiPrepared": true,
            "position": "absolute",
            "transform": "scaleX(0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <br />
  <br />
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": 2,
        "boxShadow": undefined,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }
  >
    <button
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgb(229, 229, 229)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "default",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": "100%",
        }
      }
      tabIndex={-1}
      type="submit"
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.3)",
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "margin": 0,
              "muiPrepared": true,
              "opacity": 1,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "userSelect": "none",
            }
          }
        >
          Sign Up
        </span>
      </div>
    </button>
  </div>
</form>
`;
